src/helpers/langrangeInterpolatePoly.ts

```diff
@@ -137,7 +137,7 @@ export function generateRandomPolynomial(ecCurve: EC, degree: number, secret?: B
     while (points[shareIndex.toString("hex", 64)] !== undefined) {
       shareIndex = generatePrivateExcludingIndexes([new BN(0)]);
     }
-    points[shareIndex.toString("hex", 64)] = new Point(shareIndex, new BN(generatePrivate()), ecCurve);
+    points[shareIndex.toString("hex", 64)] = new Point(shareIndex, new BN(generatePrivate()).umod(ecCurve.curve.n), ecCurve);
```

Comment:
No, it outputs a buffer

Reply:
Ah, wait, so we still use the `generatePrivate()` from `@toruslabs/eccrypto`. This is actually a problem because it generates `secp256k1` scalars if I recall correctly. Should use `ecCurve.genKeyPair().getPrivate()` or something similar instead.

This needs to be checked for all usages of `generatePrivate()` from `@toruslabs/eccrypto`.

Reply:
thanks for noticing this, yeah this is quite a important point. I will use the elliptic lib to generate priv key rather thn  @toruslabs/eccrypto

Reply:
resolved, generating priv keys with elliptic package now